fix(Pulsar): Support Apache Pulsar 4

entvex opened the pull request:
## What does this PR do?
- Replaced deprecated script arguments with the updated ones to ensure proper configuration handling.
- Upgraded the Pulsar image to version **3.0.8**.
- Updated default file paths to align with version **4.0.X** for better compatibility and usability.

## Why is it important?
The new Apache Pulsar **4.0.X** container image enforces stricter security by reducing user privileges. As a result, the `secret.key` file needs to be created in a location that can be written to in both **3.0.X** and **4.0.X** versions.
This change ensures **backward compatibility**, which is a core principle of Testcontainers, while also taking advantage of the latest improvements in Pulsar **4.0.X**.

## How to test this PR
- Ran the test suite using Apache Pulsar broker images for both versions **3.0.8** and **4.0.2** to successfully validate compatibility across both versions.
- Tests passed in both scenarios.

### Open Question
@HofmeisterAn, would you prefer that I configure the tests to explicitly validate against both image versions (e.g., **3.0.X** and **4.0.X**), or is it sufficient to test exclusively with the latest default version (`3.0.X`)? Let me know how you'd like to proceed.

- Closes #1291
- Closes #1294

Commit:
Update PulsarContainer script and defaults for compatibility

Replaced deprecated script arguments with updated ones to ensure proper configuration handling. Updated Pulsar image to version 3.0.8 and modified default file paths for better alignment with expected conventions.

HofmeisterAn commented on src/Testcontainers.Pulsar/PulsarBuilder.cs:
```diff
@@ -4,15 +4,15 @@ namespace Testcontainers.Pulsar;
 public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
 {
-    public const string PulsarImage = "apachepulsar/pulsar:3.0.6";
+    public const string PulsarImage = "apachepulsar/pulsar:3.0.8";
```
Do the changes require bumping the patch version?

entvex replied:
> Do the changes require bumping the patch version?

It's not strictly required. But I would advise we do bump the patch version. It's also [backwards compatible](https://pulsar.apache.org/contribute/release-policy/#compatibility-between-releases) so it won't create any issues.

HofmeisterAn reviewed (commented):
Thanks for the PR! It looks like this PR relates to https://github.com/testcontainers/testcontainers-dotnet/pull/1294 but with fewer necessary changes 👍.

> @HofmeisterAn, would you prefer that I configure the tests to explicitly validate against both image versions (e.g., **3.0.X** and **4.0.X**), or is it sufficient to test exclusively with the latest default version (`3.0.X`)? Let me know how you'd like to proceed.

Usually, we test both versions if they are not compatible. We follow this pattern, which should be easy to apply to `PulsarContainerTest`. I believe we just need to inherit from `PulsarContainerTest`:

https://github.com/testcontainers/testcontainers-dotnet/blob/888da44936457c0d3dea0d149e4fb61f23544b33/tests/Testcontainers.MongoDb.Tests/MongoDbContainerTest.cs#L101-L108

Extending the tests would be great.

entvex commented:
> Thanks for the PR! It looks like this PR relates to #1294 but with fewer necessary changes 👍.
> 
> > @HofmeisterAn, would you prefer that I configure the tests to explicitly validate against both image versions (e.g., **3.0.X** and **4.0.X**), or is it sufficient to test exclusively with the latest default version (`3.0.X`)? Let me know how you'd like to proceed.
> 
> Usually, we test both versions if they are not compatible. We follow this pattern, which should be easy to apply to `PulsarContainerTest`. I believe we just need to inherit from `PulsarContainerTest`:
> 
> https://github.com/testcontainers/testcontainers-dotnet/blob/888da44936457c0d3dea0d149e4fb61f23544b33/tests/Testcontainers.MongoDb.Tests/MongoDbContainerTest.cs#L101-L108
> 
> Extending the tests would be great.

Sure, I'll do that. So we know that the two LTS versions work.

Commit:
Update Pulsar image version and add new LTS 4.0.2 tests

Update the default Pulsar image to version 3.0.9 in the builder. Introduce two new test cases using Pulsar LTS 4.0.2: one for default configuration and another for authenticated configuration.

HofmeisterAn reviewed (approved):
Thanks 🙌.
